src/util/BlankNodeManager.h

```diff
@@ -91,25 +92,41 @@ class BlankNodeManager {
     // BlankNodeIndex blocks.
     template <std::ranges::range R>
     void mergeWith(const R& localBlankNodeManagers) {
-      std::ranges::copy_if(
-          localBlankNodeManagers, std::back_inserter(otherManagers_),
-          [](const std::shared_ptr<const LocalBlankNodeManager>& l) -> bool {
-            return l != nullptr;
-          });
+      auto inserter = std::back_inserter(otherBlocks_);
+      for (const auto& l : localBlankNodeManagers) {
+        if (l == nullptr) {
+          continue;
+        }
+        std::ranges::copy(l->otherBlocks_, inserter);
+        *inserter = l->blocks_;
+      }
     }
 
+    // Getter for the `blankNodeManager_` pointer required in
+    // `LocalVocab::mergeWith`.
+    BlankNodeManager* blankNodeManager() const { return blankNodeManager_; }
+
+   private:
     // Reference to the BlankNodeManager, used to free the reserved blocks.
     BlankNodeManager* blankNodeManager_;
 
-   private:
     // Reserved blocks.
-    std::vector<BlankNodeManager::Block> blocks_;
+    using Blocks = std::vector<BlankNodeManager::Block>;
+    std::shared_ptr<Blocks> blocks_{
+        new Blocks(), [this](auto blocksPtr) {
+          auto ptr = blankNodeManager_->usedBlocksSet_.wlock();
```

Comment:
```suggestion
        new Blocks(), [blankNodeManager = balnkNodeManager()](auto blocksPtr) {
          auto ptr = blankNodeManager->usedBlocksSet_.wlock();
```
`this` is dangling if the blocks are merged into another LBNM which lives longer.